plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsListenersTest.kt

```diff
@@ -273,7 +273,7 @@ class CameraAnimationsListenersTest {
     bearingAnimator.start()
     Shadows.shadowOf(Looper.getMainLooper()).idle()
 
-    Assert.assertEquals(5, valuesList.size)
+    Assert.assertEquals(3, valuesList.size)
```

Comment:
The number of triggered events has been reduced here, because I've changed the `userUpdateListeners` from mutable list to `CopyOnWriteArraySet`.
As we are adding the same user update listeners twice, I believe the correct behaviour is to ignore the second addlistener, as it will result in duplicated update events. I'd consider this maybe a bug fix instead, wdyt? cc @kiryldz @yunikkk

Reply:
Yeah I agree it should be correct to deliver the animated value once to the same listener. However I cant say I understand the test fully. Why is it 3 now since? Does it send start and end value for the animation with duration 0?

Reply:
to be honest I've no idea why it was 5 before, maybe @kiryldz have more context on it?

Reply:
why is adding the same listener considered to be a bug?

Reply:
Adding same listener twice effective means for each camera update, user will be notified twice, is that expected though? What's the use case for this behaviour?

Reply:
And the listeners behaviour I think so far is aligned across the codebase, do you see issues with that approach?